src/Utilities/Tuple.hpp

```diff
@@ -174,3 +176,135 @@ constexpr inline void tuple_transform(
       tuple, std::forward<N_aryOp>(op),
       std::make_index_sequence<sizeof...(Elements)>{}, args...);
 }
+
+namespace detail {
+/*!
+ * \brief counts the arguments of a function object.
+ */
+template <typename T>
+struct count_arg;
+
+template <typename R, typename... Args>
+struct count_arg<R(Args...)> {
+  static const size_t value = sizeof...(Args);
+};
+}  // namespace detail
+
+//{@
+/*!
+ * \ingroup UtilitiesGroup
+ * \brief Perform a function over combinatoric subsets of a std::tuple
+ *
+ * \details
+ * If given a function object, the apply_tuple_combinations function will
+ * apply the function to all ordered combinations. For instance, if the tuple
+ * contains elements (a,b,c) and the function f accepts two arguments it will be
+ * executed on all nine combinations of the three possible
+ * arguments. Importantly, the function f should then be valid for all such
+ * argument combinations.
+ *
+ * Two forms of the function are provided, one is expected to be executed with a
+ * std::function as the first argument, and a second form of the function which
+ * does not constrain the function argument but requires an explicit template
+ * parameter for the number of arguments the function takes. This second form is
+ * particularly useful for lambdas.
+ *
+ * \param set - the tuple for which you'd like to execute every combination
+ *
+ * \param function - when ArgLength is not specified, must be a
+ * std::function. If ArgLength is specified, any callable may be used.
+ *
+ * \param args - a set of partially assembled arguments passed recursively
+ *
+ * Without argument length specification, the function is called as
+ * \snippet Utilities/Test_Tuple.cpp tuple_combos_cast
+ *
+ * With argument length specification, the function is called as
+ * \snippet Utilities/Test_Tuple.cpp tuple_combos_lambda
+ */
+template <typename T, typename... Elements, typename... Args,
+          Requires<sizeof...(Args) == tmpl::size<typename tt::function_info<
+                                          T>::argument_types>::value> = nullptr>
+constexpr inline void apply_tuple_combinations(
+    const std::tuple<Elements...>& set, const std::function<T>& function,
```

Comment:
But you're still using `std::function`. Can't you not use that?

Reply:
oops, gitfail. fixing...